src/ampycloud/utils/utils.py

```diff
@@ -150,6 +159,23 @@ def check_data_consistency(pdf: pd.DataFrame,
             logger.warning('Dropping the superfluous %s column from the input data.', key)
             data.drop((key), axis=1, inplace=True)
 
+    # Check for any duplicated entry, which would make no sense.
+    if (duplic := data.duplicated()).any():
+        raise AmpycloudError('Duplicated hits in the input data:\n'
+                             f'{data[duplic].to_string(index=False)}')
+
+    # Check for inconsistencies
```

Comment:
I wanted to make sure to understand what is being done here and tried to reproduce. For me, the suggestion below is a bit more readable as I'm not so fluent with the merge command. But it's likely a matter of taste and I leave the choice to you @fpavogt .

Reply:
See comment below - proposition is now equivalent to original code.